new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSessionMigration.java

```diff
@@ -0,0 +1,462 @@
+                        .nodePort(0))))
+            .addClustersItem(new Cluster()
+                .clusterName(clusterName)
+                .replicas(replicaCount)
+                .serverStartState("RUNNING"))
+            .configuration(new Configuration()
+                .model(new Model()
+                    .domainType("WLS")
+                    .runtimeEncryptionSecret(encryptionSecretName))
+                .introspectorJobActiveDeadlineSeconds(300L)));
+
+    // create domain using model in image
+    logger.info("Create model in image domain {0} in namespace {1} using docker image {2}",
+        domainUid, domainNamespace, miiImage);
+    createDomainAndVerify(domain, domainNamespace);
+  }
+
+  private void shutdownServerUsingServerStartPolicy(String msName) {
+    final String podName = domainUid + "-" + msName;
+
+    // shutdown a server by changing the it's serverStartPolicy property.
+    logger.info("Shutdown the server {0}", msName);
+    boolean serverStopped = assertDoesNotThrow(() ->
+        shutdownManagedServerUsingServerStartPolicy(domainUid, domainNamespace, msName));
+    assertTrue(serverStopped,
+        String.format("Failed to shutdown server %s ", msName));
+
+    // check that the managed server pod shutdown successfylly
+    logger.info("Check that managed server pod {0} stopped in namespace {1}",
+        podName, domainNamespace);
+    checkPodDoesNotExist(podName, domainUid, domainNamespace);
+  }
+
+  private static String buildCurlCommand(String podName,
+                                         String curlUrlPath,
+                                         String headerOption) {
+    logger.info("Build a curl command with pod name {0}, curl URL path {1} and HTTP header option {2}",
+        podName, curlUrlPath, headerOption);
+    final String httpHeaderFile = "/u01/oracle/header";
+
+    StringBuffer curlCmd = new StringBuffer("curl --silent --show-error http://");
+    curlCmd.append(podName)
+        .append(":")
+        .append(managedServerPort)
+        .append("/")
+        .append(curlUrlPath)
+        .append(headerOption)
+        .append(httpHeaderFile);
+
+    return curlCmd.toString();
+  }
+
+  private static Map<String, String> processHttpRequest(String serverName,
+                                                        String curlUrlPath,
+                                                        String headerOption) {
+    String[] httpAttrArray =
+        {"sessioncreatetime", "sessionid", "primary", "secondary", "count"};
+    Map<String, String> httpAttrInfo = new HashMap<String, String>();
+
+    // build curl command
+    String curlCmd = buildCurlCommand(serverName, curlUrlPath, headerOption);
+    logger.info("Command to set HTTP request and get HTTP response {0} ", curlCmd);
+
+    // set HTTP request and get HTTP response
+    ExecResult execResult = assertDoesNotThrow(
+        () -> execCommand(domainNamespace, adminServerPodName,
```

Comment:
just a question - if you use node port and cluster url for app, you don't need to exec into the pod to curl?

Reply:
good point. I wanted to use cluster node port but I can't find cluster node port, it's not enabled in our framework. I wanted to add cluster node port but not sure the right way